newswires/app/db/FingerpostWireEntry.scala

```diff
@@ -106,23 +115,37 @@ object FingerpostWireEntry extends SQLSyntaxSupport[FingerpostWireEntry] {
       .list()
       .apply()
 
-    QueryResponse(results)
+    val keywordCounts = getKeywords(additionalWhereClauses =
+      whereClauses
+    ) // TODO do this in parallel
+
+    QueryResponse(results, keywordCounts)
   }
 
-  def getKeywords(maybeInLastHours: Option[Int], maybeLimit: Option[Int]) =
+  def getKeywords(
+      maybeInLastHours: Option[Int] = None,
+      maybeLimit: Option[Int] = None,
+      additionalWhereClauses: List[SQLSyntax] = Nil
+  ) =
     DB readOnly { implicit session =>
-      val innerWhereClause = maybeInLastHours
-        .fold(sqls"")(inLastHours =>
-          sqls"WHERE ingested_at > now() - ($inLastHours::text || ' hours')::interval"
-        )
+      val innerWhereClause = additionalWhereClauses ++ maybeInLastHours
+        .map(inLastHours =>
+          sqls"ingested_at > now() - ($inLastHours::text || ' hours')::interval"
+        ) match {
+        case Nil        => sqls""
+        case whereParts => sqls"WHERE ${sqls.joinWithAnd(whereParts: _*)}"
+      }
+
       val limitClause = maybeLimit
         .map(limit => sqls"LIMIT $limit")
         .orElse(maybeInLastHours.map(_ => sqls"LIMIT 10"))
         .getOrElse(sqls"")
       sql"""| SELECT distinct keyword, count(*)
             | FROM (
-            |     SELECT jsonb_array_elements(content -> 'keywords') as keyword
-            |     FROM fingerpost_wire_entry
+            |     SELECT jsonb_array_elements(${FingerpostWireEntry.syn.column(
```

Comment:
not introduced in this PR, but all keywords come back with an extra level of stringification
<img width="331" alt="image" src="https://github.com/user-attachments/assets/c87fc815-d341-4200-8ef1-47aee058a5df">

should this be `jsonb_array_elements_text`? (or, do an extra parse on the server?)

Reply:
ooo good spot @andrew-nowak, my JSON view extension had obscured that